ledger/tests/helpers/mod.rs

```diff
@@ -0,0 +1,290 @@
+        let genesis_rng = &mut TestRng::from_seed(seed);
+        let private_keys = (0..committee_size).map(|_| PrivateKey::new(genesis_rng).unwrap()).collect();
+
+        Self::from_genesis(private_keys, genesis_block)
+    }
+
+    /// Initialize the builder with the specified commitee and gensis block
+    pub fn from_genesis(private_keys: Vec<PrivateKey<CurrentNetwork>>, genesis_block: Block<CurrentNetwork>) -> Self {
+        // Initialize the ledger with the genesis block.
+        let ledger = Ledger::<CurrentNetwork, LedgerType<CurrentNetwork>>::load(
+            genesis_block.clone(),
+            StorageMode::new_test(None),
+        )
+        .unwrap();
+
+        Self {
+            private_keys,
+            ledger,
+
+            genesis_block,
+            current_height: 1,
+            last_batch: Default::default(),
+            last_committed_batch: Default::default(),
+            last_block_round: 1,
+            round_to_certificates: Default::default(),
+            previous_leader_certificate: Default::default(),
+        }
+    }
+
+    /// Create multiple blocks, with fully-connected DAGs.
+    #[allow(dead_code)]
+    pub fn generate_blocks(&mut self, num_blocks: usize, rng: &mut TestRng) -> Vec<Block<CurrentNetwork>> {
+        self.generate_blocks_with_opts(num_blocks, &BlockOptions::default(), rng)
+    }
+
+    /// Create multiple blocks, with additional parameters
+    pub fn generate_blocks_with_opts(
+        &mut self,
+        num_blocks: usize,
+        options: &BlockOptions,
+        rng: &mut TestRng,
+    ) -> Vec<Block<CurrentNetwork>> {
+        assert!(num_blocks > 0, "Need to build at least one block");
+
+        (0..num_blocks).map(|_| self.generate_block_with_opts(options, rng)).collect()
+    }
+
+    /// Create a new block, with a fully-connected DAG.
+    ///
+    /// This will "fill in " any gaps left in earlier rounds from non participating nodes.
+    pub fn generate_block(&mut self, rng: &mut TestRng) -> Block<CurrentNetwork> {
+        self.generate_block_with_opts(&BlockOptions::default(), rng)
+    }
+
+    /// Same as `generate_block` but with additional options/parameters.
+    pub fn generate_block_with_opts(&mut self, options: &BlockOptions, rng: &mut TestRng) -> Block<CurrentNetwork> {
+        assert!(self.current_height > 0);
+        assert!(options.skip_nodes.len() * 3 < self.private_keys.len());
+
+        let block_height = self.current_height + 1;
+
+        // SubDAGs can be at most GC roudns long.
+        let mut round = if self.last_block_round < BatchHeader::<CurrentNetwork>::MAX_GC_ROUNDS as u64 {
+            1
+        } else {
+            self.last_block_round - BatchHeader::<CurrentNetwork>::MAX_GC_ROUNDS as u64 + 2
+        };
+
+        // Create certificates for each round.
+        loop {
+            let mut created_anchor = false;
+
+            let previous_certificate_ids = if round <= 1 {
```

Comment:
```suggestion
            let previous_certificate_ids = if round == 1 {
```
Since `round` is never initialized to 0, a few lines above in the code.